src/Featurizers/UnitTests/LagLeadOperatorFeaturizer_UnitTests.cpp

```diff
@@ -0,0 +1,651 @@
+// ----------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License
+// ----------------------------------------------------------------------
+#define CATCH_CONFIG_MAIN
```

Comment:
>CATCH_CONFIG_MAIN [](start = 8, length = 17)

I believe that all of these tests invoke the functionality with deltas provided in sorted order; can you add a test where the incoming deltas are not in sorted order to verify things still work as expected.

Reply:
Added a test that deltas are not in sorted order